tests/test_classifier/test_confusion_matrix.py

```diff
@@ -210,38 +211,13 @@ def test_class_filter_eg_zoom_in(self):
 
     def test_extra_classes(self):
         """
-        Assert that any extra classes are simply ignored
+        Assert that any extra classes raise an exception
         """
-        # TODO: raise exception instead
-        _, ax = plt.subplots()
-
         model = LogisticRegression(random_state=93)
-        cm = ConfusionMatrix(model, ax=ax, classes=[0, 1, 2, 11])
-        cm.fit(self.digits.X.train, self.digits.y.train)
-        cm.score(self.digits.X.test, self.digits.y.test)
-
-        npt.assert_array_equal(cm.class_counts_, [38, 37, 39, 0])
-
-        # Ensure correct confusion matrix under the hood
-        npt.assert_array_equal(
-            cm.confusion_matrix_,
-            np.array([[38, 0, 0, 0], [0, 35, 0, 0], [0, 0, 39, 0], [0, 0, 0, 0]]),
-        )
+        cm = ConfusionMatrix(model, classes=[0, 1, 2, 11])
 
-        self.assert_images_similar(cm, tol=10)
-
-    def test_one_class(self):
```

Comment:
Not sure this test makes sense anymore; but I could try to do it with a OneClassSVM and use force_model=True. Happy for comments on this.